Dockerfile

```
@@ -0,0 +1,185 @@
+    make -j $(nproc) install && \
+    ldconfig
+
+ARG BIOBAMBAM2_VERSION
+RUN curl -sSL -O "https://gitlab.com/german.tischler/biobambam2/-/archive/${BIOBAMBAM2_VERSION}/biobambam2-${BIOBAMBAM2_VERSION}.tar.bz2" && \
+    tar xfj biobambam2-${BIOBAMBAM2_VERSION}.tar.bz2 && \
+    cd biobambam2-${BIOBAMBAM2_VERSION} && \
+    ./configure && \
+    make -j $(nproc) install
+
+ARG TEEPOT_VERSION
+RUN curl -sSL -O "https://github.com/wtsi-npg/teepot/releases/download/${TEEPOT_VERSION}/teepot-${TEEPOT_VERSION}.tar.gz" && \
+    tar xzf teepot-${TEEPOT_VERSION}.tar.gz && \
+    cd teepot-${TEEPOT_VERSION} && \
+    ./configure && \
+    make -j $(nproc) install
+
+ARG HTSLIB_VERSION
+RUN curl -sSL -O "https://github.com/samtools/htslib/releases/download/${HTSLIB_VERSION}/htslib-${HTSLIB_VERSION}.tar.bz2" && \
+    tar xfj htslib-${HTSLIB_VERSION}.tar.bz2 && \
+    cd htslib-${HTSLIB_VERSION} && \
+    ./configure --with-libdeflate --enable-libcurl --enable-s3 && \
+    make -j $(nproc) install && \
+    ldconfig
+
+ARG SAMTOOLS_VERSION
+RUN curl -sSL -O "https://github.com/samtools/samtools/releases/download/${SAMTOOLS_VERSION}/samtools-${SAMTOOLS_VERSION}.tar.bz2" && \
+    tar xfj samtools-${SAMTOOLS_VERSION}.tar.bz2 && \
+    cd samtools-${SAMTOOLS_VERSION} && \
+    ./configure --with-htslib=system --without-curses && \
+    make -j $(nproc) install
+
+ARG BWA_VERSION
+RUN curl -sSL -O "https://github.com/lh3/bwa/archive/refs/tags/v${BWA_VERSION}.tar.gz" && \
+    tar xzf v${BWA_VERSION}.tar.gz && \
+    cd bwa-${BWA_VERSION} && \
+    make -j $(nproc) && \
+    cp ./bwa /usr/local/bin/ && \
+    chmod +x /usr/local/bin/bwa
+
+ARG BAMBI_VERSION
+RUN git clone --single-branch --branch="$BAMBI_VERSION" --depth=1 "https://github.com/wtsi-npg/bambi.git" && \
+    cd bambi && \
+    autoreconf -fi && \
+    ./configure && \
+    make -j $(nproc) install
+
+ARG NPG_SEQ_COMMON_VERSION
+RUN git clone --single-branch --branch="$NPG_SEQ_COMMON_VERSION" --depth=1 "https://github.com/wtsi-npg/npg_seq_common.git" && \
+    cd npg_seq_common && \
+    cp ./bin/seqchksum_merge.pl /usr/local/bin/ && \
+    chmod +x /usr/local/bin/seqchksum_merge.pl
+
+RUN cpanm --notest --local-lib /usr/local/lib Module::Build
+
+COPY . ./p4
+
+RUN cd p4 && \
+    cpanm --notest --local-lib /usr/local/lib --installdeps . && \
+    cpanm --notest --local-lib /usr/local/lib .
+
+
+FROM $BASE_IMAGE
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    apt-utils \
+    ca-certificates \
+    lsb-release \
+    unattended-upgrades \
+    locales && \
+    locale-gen en_GB en_GB.UTF-8 && \
+    localedef -i en_GB -c -f UTF-8 -A /usr/share/locale/locale.alias en_GB.UTF-8
+
+ENV LANG=en_GB.UTF-8 \
+    LANGUAGE=en_GB \
+    LC_ALL=en_GB.UTF-8
+
+RUN apt-get install -q -y --no-install-recommends \
+        libboost-atomic1.74.0 \
+        libbz2-1.0 \
+        libcurl4 \
+        liblzma5 \
```

```suggestion
        libcurl4-gnutls-dev \
        libgomp1 \
        liblzma5 \
```

`bamseqchksum` fails to run in final container without these.